components/controller.ts

```diff
@@ -857,6 +861,8 @@ export class Controller {
                 log(LogLevel.DEBUG, e, "contracts")
             }
         }
+
+        log(LogLevel.INFO, "Contracts loading completed", "contracts")
```

Comment:
Maybe it would be best to do this as a `.then` on the Promise that `index` returns so that it's more accurate?

Reply:
Didn't notice that, good point. I'll add edit it now.

Reply:
This shouldn't matter, no? In both cases, the log is called after all promises are awaited

Reply:
The other promises are blocked on their awaits, but just because they are guaranteed to be done before the first one doesn't mean the first one isn't.